toolchain/parse/typed_nodes.h

```diff
@@ -1032,6 +1034,51 @@ struct DesignatorExpr {
   NodeIdOneOf<IdentifierName, SelfTypeName> name;
 };
 
+// FIXME: RequirementRewrite?
+struct RequirementAssign {
+  static constexpr auto Kind = NodeKind::RequirementAssign.Define(
+      {.category = NodeCategory::Requirement, .child_count = 2});
+  AnyExprId lhs;
+  Lex::EqualTokenIndex token;
+  AnyExprId rhs;
+};
+
+// FIXME: RequirementEqualEqual?
+struct RequirementEquals {
+  static constexpr auto Kind = NodeKind::RequirementEquals.Define(
+      {.category = NodeCategory::Requirement, .child_count = 2});
+  AnyExprId lhs;
+  Lex::EqualEqualTokenIndex token;
+  AnyExprId rhs;
+};
+
+struct RequirementImpls {
+  static constexpr auto Kind = NodeKind::RequirementImpls.Define(
+      {.category = NodeCategory::Requirement, .child_count = 2});
+  AnyExprId lhs;
+  Lex::ImplsTokenIndex token;
+  AnyExprId rhs;
+};
+
+// virtual node
```

Comment:
Seems like a good idea to use the same commenting style here and in `ShortCircuit{And,Or}Operand` when mentioning that this is a virtual node.

Reply:
I've made that change, but I didn't really understand those comments. They talk about how the `and`/`or` token is owned by one of the nodes, but nothing seems to distinguish which node is virtual and which owns the token.

Reply:
Yeah, I think the only thing that distinguishes which node "owns" the token is these comments :) And it seems to be an arbitrary choice that doesn't really impact anything -- I think we could equally say that the nodes share the token. But I think if there were a difference, it'd be: if a tool wants to map from a token to a parse node, the "non-virtual" one is the one they should find, which I think means the non-virtual node should always be a parent of the virtual node, so that they can both be discovered by tooling. Maybe that should be the rule we use: virtual parse nodes for a token should always be children of the non-virtual parse node.